tests/director/tests/formation_api_test.go

```diff
@@ -695,6 +698,134 @@ func TestSubaccountInAtMostOneFormationOfType(t *testing.T) {
+
+	t.Logf("Should create second formation: %s", secondFormationName)
+	secondFormationInput := fixtures.FixFormationInput(secondFormationName, str.Ptr(formationTemplateName))
+	secondFormationInputGQL, err := testctx.Tc.Graphqlizer.FormationInputToGQL(secondFormationInput)
+	require.NoError(t, err)
+
+	var secondFormation graphql.Formation
+	defer fixtures.DeleteFormation(t, ctx, certSecuredGraphQLClient, secondFormationName)
+	createSecondReq := fixtures.FixCreateFormationWithTemplateRequest(secondFormationInputGQL)
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, createSecondReq, &secondFormation)
+	require.NoError(t, err)
+	require.Equal(t, secondFormationName, secondFormation.Name)
+
+	t.Log("Create first application")
+	app := graphql.ApplicationExt{} // needed so the 'defer' can be above the application creation
+	defer fixtures.CleanupApplication(t, ctx, certSecuredGraphQLClient, tenantId, &app)
+	app, err = fixtures.RegisterApplicationWithApplicationType(t, ctx, certSecuredGraphQLClient, "e2e-tests-app", conf.ApplicationTypeLabelKey, string(util.ApplicationTypeC4C), tenantId)
+	require.NoError(t, err)
+	require.NotEmpty(t, app.ID)
+
+	t.Log("Create second application with exception type")
+	exceptionTypeApp := graphql.ApplicationExt{} // needed so the 'defer' can be above the application creation
+	defer fixtures.CleanupApplication(t, ctx, certSecuredGraphQLClient, tenantId, &exceptionTypeApp)
+	exceptionTypeApp, err = fixtures.RegisterApplicationWithApplicationType(t, ctx, certSecuredGraphQLClient, "e2e-tests-exceptionType-app", conf.ApplicationTypeLabelKey, exceptionSystemType, tenantId)
+	require.NoError(t, err)
+	require.NotEmpty(t, exceptionTypeApp.ID)
+
+	t.Logf("Assign first application to first formation with name: %s", firstFormation.Name)
+	defer fixtures.UnassignFormationWithApplicationObjectType(t, ctx, certSecuredGraphQLClient, graphql.FormationInput{Name: firstFormation.Name}, app.ID, tenantId)
+	assignReq := fixtures.FixAssignFormationRequest(app.ID, string(graphql.FormationObjectTypeApplication), firstFormation.Name)
+	var assignFormation graphql.Formation
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, assignReq, &assignFormation)
+	require.NoError(t, err)
+	require.Equal(t, firstFormation.Name, assignFormation.Name)
+
+	t.Logf("Should fail to assign first application to another formation with name: %s of the same kind", secondFormation.Name)
+	defer fixtures.UnassignFormationWithApplicationObjectType(t, ctx, certSecuredGraphQLClient, graphql.FormationInput{Name: secondFormation.Name}, app.ID, tenantId)
+	assignReq = fixtures.FixAssignFormationRequest(app.ID, string(graphql.FormationObjectTypeApplication), secondFormation.Name)
+	err = testctx.Tc.RunOperation(ctx, certSecuredGraphQLClient, assignReq, &assignFormation)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "Operator \"IsNotAssignedToAnyFormationOfType\" is not satisfied")
```

Comment:
We should use the constant IsNotAssignedToAnyFormationOfTypeOperator for the operator name

Reply:
Not applicable because the fixture that expects error doesn't check the actual error.